pageserver/src/tenant/storage_layer.rs

```diff
@@ -138,12 +186,56 @@ impl Default for LayerAccessStatsInner {
             count_by_access_kind: EnumMap::default(),
             task_kind_flag: EnumSet::default(),
             last_accesses: VecDeque::with_capacity(LayerAccessStats::LAST_ACCESSES_MAX_LEN),
+            last_residence_changes: VecDeque::with_capacity(
+                LayerAccessStats::LAST_RESIDENCE_CHANGES_MAX_LEN,
+            ),
         }
     }
 }
 
 impl LayerAccessStats {
     const LAST_ACCESSES_MAX_LEN: usize = 16;
+    const LAST_RESIDENCE_CHANGES_MAX_LEN: usize = 16;
```

Comment:
Optional: If you plan to keep these elements const, you can go down to arrays, there's https://doc.rust-lang.org/std/primitive.slice.html#method.rotate_right to help with the buffer.

Reply:
I think the correct abstraction would be https://docs.rs/heapless/latest/heapless/struct.HistoryBuffer.html

Not sure I want to drag that library in, though :)

Reply:
TIL: HistoryBuffer, even though I try to peek every now and then what's in heapless.

Reply:
Pushed a change that switches to HistoryBuffer to this branch. @SomeoneToIgnore  what do you think?

Reply:
A bit sad that there's no release for that, but if you like `HistoryBuffer` more, why not.